presto-base-arrow-flight/src/test/java/com/facebook/plugin/arrow/testingServer/TestingArrowProducer.java

```diff
@@ -0,0 +1,310 @@
+import org.apache.arrow.vector.VectorLoader;
+import org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.arrow.vector.VectorUnloader;
+import org.apache.arrow.vector.ipc.message.ArrowRecordBatch;
+import org.apache.arrow.vector.types.DateUnit;
+import org.apache.arrow.vector.types.FloatingPointPrecision;
+import org.apache.arrow.vector.types.TimeUnit;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.types.pojo.Schema;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.TimeZone;
+import java.util.concurrent.ThreadLocalRandom;
+
+import static com.facebook.airlift.json.JsonCodec.jsonCodec;
+import static com.facebook.presto.common.Utils.checkArgument;
+import static org.apache.arrow.adapter.jdbc.JdbcToArrowUtils.jdbcToArrowSchema;
+
+public class TestingArrowProducer
+        implements FlightProducer
+{
+    private final BufferAllocator allocator;
+    private final Connection connection;
+    private static final Logger logger = Logger.get(TestingArrowProducer.class);
+    private final JsonCodec<TestingArrowFlightRequest> requestCodec;
+    private final JsonCodec<TestingArrowFlightResponse> responseCodec;
+
+    public TestingArrowProducer(BufferAllocator allocator) throws Exception
+    {
+        this.allocator = allocator;
+        String h2JdbcUrl = "jdbc:h2:mem:testdb" + System.nanoTime() + "_" + ThreadLocalRandom.current().nextInt() + ";DB_CLOSE_DELAY=-1";
+        TestingH2DatabaseSetup.setup(h2JdbcUrl);
+        this.connection = DriverManager.getConnection(h2JdbcUrl, "sa", "");
+        this.requestCodec = jsonCodec(TestingArrowFlightRequest.class);
+        this.responseCodec = jsonCodec(TestingArrowFlightResponse.class);
+    }
+
+    @Override
+    public void getStream(CallContext callContext, Ticket ticket, ServerStreamListener serverStreamListener)
+    {
+        try (Statement stmt = connection.createStatement()) {
+            TestingArrowFlightRequest request = requestCodec.fromJson(ticket.getBytes());
+            checkArgument(request != null, "Request is null");
+            checkArgument(request.getQuery().isPresent(), "Query is missing");
+
+            // Extract and validate the SQL query
+            String query = request.getQuery().get();
+            if (query.trim().isEmpty()) {
+                throw new IllegalArgumentException("Query cannot be empty.");
+            }
+
+            logger.info("Executing query: " + query);
+            query = query.toUpperCase(); // Optionally, to maintain consistency
```

Comment:
This comment doesn't appear to mean anything?

Reply:
I'm not really sure. I'll remove the comment and keep the uppercase conversion, in case that is preferred by the h2 driver.